client/components/domains/use-your-domain-step/index.tsx

```diff
@@ -0,0 +1,313 @@
+									{ phrase }
+								</div>
+							);
+						} ) }
+					</div>
+				</div>
+				<div className="use-your-domain-step__option-action">
+					<Button
+						className="use-your-domain-step__option-button"
+						primary={ isPrimary }
+						onClick={ onClick }
+					>
+						{ buttonText }
+					</Button>
+					<div className="use-your-domain-step__learn-more">{ learnMore }</div>
+				</div>
+			</div>
+		</Card>
+	);
+}
+
+type UseYourDomainStepProps = {
+	basePath: string;
+	goBack?: () => void;
+	initialQuery?: string;
+};
+
+function UseYourDomainStep( { basePath, goBack, initialQuery }: UseYourDomainStepProps ) {
+	const cartKey = useCartKey();
+	const { responseCart: cart } = useShoppingCart( cartKey );
+
+	const translate = useTranslate();
+	const dispatch = useDispatch();
+
+	const [ searchQuery ] = useState( initialQuery || '' );
+
+	const selectedSite = useSelector( getSelectedSite );
+	const productsList = useSelector( getProductsList );
+	const currencyCode = useSelector( getCurrentUserCurrencyCode );
+	const domainsWithPlansOnly = useSelector( ( state ) =>
+		currentUserHasFlag( state, DOMAINS_WITH_PLANS_ONLY )
+	);
+	const primaryWithPlansOnly = useSelector( ( state ) =>
+		currentUserHasFlag( state, NON_PRIMARY_DOMAINS_TO_FREE_USERS )
+	);
+
+	const domainsWithPlansOnlyButNoPlan =
+		domainsWithPlansOnly && selectedSite?.plan && ! isPlan( selectedSite.plan );
+	const productSlug = getDomainProductSlug( searchQuery );
+	const domainProductSalePrice = getDomainTransferSalePrice(
+		productSlug,
+		productsList,
+		currencyCode
+	);
+
+	let domainProductFreeText = null;
+	if ( isNextDomainFree( cart ) || isDomainBundledWithPlan( cart, searchQuery ) ) {
+		domainProductFreeText = translate( 'Free with your plan' );
+	} else if ( domainsWithPlansOnlyButNoPlan ) {
+		domainProductFreeText = translate( 'Included in paid plans' );
+	}
+
+	let transferSalePriceText = null;
+	if (
+		domainProductSalePrice !== null &&
+		! isNextDomainFree( cart ) &&
+		! isDomainBundledWithPlan( cart, searchQuery ) &&
+		domainsWithPlansOnlyButNoPlan
+	) {
+		transferSalePriceText = translate( 'Sale price is %(cost)s', {
+			args: { cost: domainProductSalePrice },
+		} );
+	}
+
+	let transferPriceText = null;
+	const domainProductPrice = getDomainPrice( productSlug, productsList, currencyCode );
+	if ( domainProductPrice ) {
+		if (
+			isNextDomainFree( cart ) ||
+			isDomainBundledWithPlan( cart, searchQuery ) ||
+			domainsWithPlansOnlyButNoPlan ||
+			getDomainTransferSalePrice( productSlug, productsList, currencyCode )
+		) {
+			transferPriceText = translate( 'Renews at %(cost)s', { args: { cost: domainProductPrice } } );
+		} else {
+			transferPriceText = translate( '%(cost)s per year', { args: { cost: domainProductPrice } } );
+		}
+	}
+
+	let mappingPriceText;
+	const price = productsList?.domain_map?.cost;
+	if ( price && currencyCode ) {
+		mappingPriceText = formatCurrency( price, currencyCode );
```

Comment:
nit: the `formatCurrency` result should be assigned to its own variable. Like `cost`.

Reply:
https://github.com/Automattic/wp-calypso/pull/101854/commits/af0065c24222dbbbe75663c76130ff513cf9e238

Totally agree, it's always hard to draw a line when doing these refactors, as there's much more that I'd like to change 😅